pingcap/tidb
# *: integrate plugin framework with TiDB

### What problem does this PR solve? 

following up https://github.com/pingcap/tidb/pull/8788, integration plugin framework into TiDB

### What is changed and how it works?

- add `plugin_dir` and `plugin_load` config
- make `loadCommonGlobalVarsSQL` be tweakable before boostrap
- load plugin before boostrap

### Check List 

Tests 

 - manual test

```
1. checkout https://github.com/lysu/slow_log_table
2. pluginpkg slow_log_table
3. load with tidb
4. check slow log table exists
```

Code changes

 - change Impl

Side effects

 - N/A

Related changes

 - N/A


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/9006)
